cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1247,10 +1247,13 @@ private boolean isMaintenanceTimeWindowsSatisfied(Supplier<Date> dateSupplier) {
 
         private int getDeploymentCaCertGeneration(Deployment dep, Ca ca) {
             int caCertGeneration = 0;
-            if (dep != null) {
+            if (dep != null && dep.getSpec().getTemplate().getMetadata().getAnnotations() != null) {
                 String depAnnotation = getCaCertAnnotation(ca);
-                caCertGeneration =
-                        Integer.parseInt(dep.getSpec().getTemplate().getMetadata().getAnnotations().get(depAnnotation));
+
+                if (dep.getSpec().getTemplate().getMetadata().getAnnotations().get(depAnnotation) != null) {
+                    caCertGeneration =
+                            Integer.parseInt(dep.getSpec().getTemplate().getMetadata().getAnnotations().get(depAnnotation));
+                }
```

Comment:
I think you can write this whole thing as a one liner:
```
if (dep != null) {
    caCertGeneration = parseInt(io.strimzi.operator.common.Util.annotations(dep).getOrDefault(depAnnotation, "0"));
}
```

Reply:
Actually, Utils takes the annotations from Deployments metadata. The original code takes it from Template metadata. So this would not work out of the box.

Reply:
Ah, that's true. Sorry.